File: backend/src/main/java/com/mhbilliards/billiards_management/entity/TableBilliardType.java
package com.mhbilliards.billiards_management.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "table_billiard_types")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class TableBilliardType extends BaseEntity {
    String name;
    String description;
    Double costPrice;
    String supplier;
    String supplierPhone;

}

File: frontend/src/types/tableBilliardType.ts
import { IBaseResponse } from "./base";

export interface ITableBilliardTypeRequest {
    name: string;
    description: string;
    costPrice: number;
    pricePerHour: number;
    supplier: string;
    supplierPhone: string;
}

export interface ITableBilliardTypeResponse
    extends ITableBilliardTypeRequest, IBaseResponse {}
